### PR TITLE
Delayed_job systemd unit

### PR DESCRIPTION
**Pull request by enricostano**

### CODE REVIEW

File: roles/webserver/templates/delayed_job.service.j2 [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+[Service]
+Type=simple
+User=openfoodnetwork
+WorkingDirectory={{ current_path }}
+Environment=RAILS_ENV={{ rails_env }}
+SyslogIdentifier=openfoodnetwork-delayed_job
+PIDFile={{ delayed_job_pid }}
+
+ExecStart=script/delayed_job
```
Comment (sauloperez) [inline roles/webserver/templates/delayed_job.service.j2]: start? stop?

**Comment by sauloperez** [comment]
What's left here @enricostano . Something I can help you with?

File: roles/webserver/templates/delayed_job.service.j2 [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Open Food Network delayed_job systemd service
+
+[Unit]
+Description=Open Food Network delayed_job
+
+[Service]
+Type=simple
```
Comment (enricostano) [inline roles/webserver/templates/delayed_job.service.j2]: Not sure if `simple` or `forking`

File: roles/webserver/templates/delayed_job.service.j2 (hunk 1 above)
Comment (sauloperez) [inline roles/webserver/templates/delayed_job.service.j2]: This line should contain `script/delayed_job start`. We're also missing an `ExecStop`

Comment (elf-pavlik) [reply roles/webserver/templates/delayed_job.service.j2]: https://www.digitalocean.com/community/tutorials/understanding-systemd-units-and-unit-files#unit-specific-section-directives
> *ExecStop=*: This indicates the command needed to stop the service. If this is not given, the process will be killed immediately when the service is stopped.

Do we need to run `script/delayed_job start` or can just let it get killed? I don't really know how it uses PIDs etc.
Possibly relevant: http://www.rubydoc.info/gems/delayed_job/4.1.4#Running_Jobs

File: roles/webserver/tasks/main.yml [hunk 3]
```diff
@@ -12,13 +12,20 @@
...
-- name: update unicorn
+- name: Enable unicorn unit
   service:
     name: "unicorn_{{ app }}"
     enabled: yes
   become: yes
-  tags: unicorn
+  tags: init
+
+- name: Enable delayed_job unit
```
Comment (elf-pavlik) [inline roles/webserver/tasks/main.yml]: It probably needs to become root just like in #144 or run as user service like in #143 
Depends on how we resolve #82

File: roles/webserver/templates/delayed_job.service.j2 (hunk 1 above)
Comment (enricostano) [reply roles/webserver/templates/delayed_job.service.j2]: We need `start` in `ExecStart` for sure, and I would add `ExecStop` just to be consistent with DelayedJob API.

File: roles/webserver/tasks/main.yml (hunk 3 above)
Comment (enricostano) [reply roles/webserver/tasks/main.yml]: Probably we'll need to become `root`

**Comment by Matt-Yorkley** [comment]
It would be great to get this systemd unit working. We had more issues today.

File: roles/webserver/tasks/main.yml (hunk 3 above)
Comment (Matt-Yorkley) [reply roles/webserver/tasks/main.yml]: With the current script for restarting it we run it as the unicorn user and it works fine...

File: roles/webserver/templates/delayed_job.service.j2 (hunk 1 above)
Comment (Matt-Yorkley) [reply roles/webserver/templates/delayed_job.service.j2]: We can consider using this under `[Service]` as well: 
```
Restart=always
RestartSec=30
```

**Comment by sauloperez** [comment]
WAT? what was it @Matt-Yorkley ??

**Comment by Matt-Yorkley** [comment]
I've pushed an extra commit for this at https://github.com/coopdevs/ofn-install/pull/4 to finish it off and make `monit` obsolete.

**Comment by sauloperez** [comment]
and how? I'm a bit curious. Was it the recovering option after 60s of being dead?

**Comment by Matt-Yorkley** [comment]
The unit has the settings `Restart=on-failure` and `RestartSec=60` which tell systemd to monitor the service and restart it when something goes wrong, so now systemd is natively doing the old job of the third-party monit service. 

I wasn't sure about the timer of 60 seconds, it's recommended to not set it too low, and the script we run as part of this service takes a little while to execute because it goes via ruby and rbenv, so I chose 60 seconds. Monit was previously checking every 120 seconds.

**Comment by sauloperez** [comment]
Should we close this one in favor of the one you opened?

**Comment by Matt-Yorkley** [comment]
Closing as now covered by #192.

**Pull request closed without merging**